pkg/cloudclient/aws/private.go

```diff
@@ -868,8 +868,20 @@ func (ac *Client) removeLoadBalancerFromMasterNodes(ctx context.Context, kclient
 	var intDNSName, intHostedZoneID, lbName string
 	for _, networkLoadBalancer := range nlbs {
 		if networkLoadBalancer.scheme == "internet-facing" {
+
+			var nlbsToDeleteArn string
+			nlbTagsMap, err := ac.getAllTagsFromLoadBalancer(networkLoadBalancer.loadBalancerArn)
+			if err != nil {
+				fmt.Printf("There was a problem getting back network load balancer tags for NLB %s: %s\n", networkLoadBalancer.loadBalancerArn, err)
+				return "", "", err
+			}
+
 			lbName = networkLoadBalancer.loadBalancerName
-			err := ac.deleteExternalLoadBalancer(networkLoadBalancer.loadBalancerArn)
+			if _, ok := nlbTagsMap["kubernetes.io/service-name"]; !ok {
```

Comment:
What if there are multiple clusters in the same AWS account and region? Is there anyway to make sure the LB matches the specific cluster?

Reply:
there is a label that gives us the cluster name. I think that should be a good hook to identify the proper cluster.